contributor-docs/adrs/adr-XXX-file-structure.md

```diff
@@ -0,0 +1,34 @@
+# ADR XXX: File structure
+
+## Status
+
+Proposed
+
+## Context
+
+Components might be more grokable if they were structured consistently
+
+## Decision
+
+```
+primer-react/
+├─ src/
+│  ├─ BreadCrumbs/
+│  │  ├─ index.ts                    // Just re-exporting?
+│  │  ├─ BreadCrumbs.tsx             // Primary component
+│  │  ├─ BreadCrumbsItem.tsx         // Subcomponent
+│  │  ├─ BreadCrumbs.mdx             // Documentation. Always .mdx, not .md
+│  │  ├─ BreadCrumbs.stories.tsx
+│  │  ├─ BreadCrumbs.test.tsx
+│  │  └─ BreadCrumbs.types.test.tsx
```

Comment:
@colebemis also talked about possibly adding a yaml file to export things like component status? Though with the mdx file already in there, we may not need it.

Reply:
Looks like Braid Design System does something similar: https://github.com/seek-oss/braid-design-system/tree/master/lib/components/Accordion

Reply:
Yes! I think I like the idea of a separate `BreadCrumbs.yml` even though we could use front matter. I imagine it might inform some unit tests and behaviors outside of docs.